ibridges/search.py

```diff
@@ -0,0 +1,54 @@
+""" Data query
+"""
+from typing import Optional
+
+from ibridges.irodsconnector import keywords as kw
+from ibridges import Session
+
+
+def search(session: Session, path: Optional[str] = None, checksum: Optional[str] = None,
+               key_vals: Optional[dict] = None) -> list[tuple[str, str, str]]:
+    """Retrieves all collections and data objects (the absolute collection path,
+    data object or collection name) to the given user-defined and system metadata.
+    By Default all accessible collections and data objects will be returned.
+    Wildcard: %
+
+    Parameters
+    ----------
+    path: str
```

Comment:
I think we should also support the IrodsPath for the path.

Reply:
I was thinking about that, but was not sure. What if someone only wants to get a list of available search results to check whether data has been labeled but does not want to do anything with the collections and data afterwards. Would it not cause too much memory usage?